File: test_data/code/python/1518-water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        emptyBottles = numBottles
        fullBottles = numBottles
        totalBottles = fullBottles
        while emptyBottles >= numExchange:
            newFullBottles = emptyBottles // numExchange
            emptyBottles = emptyBottles % numExchange + newFullBottles
            fullBottles += newFullBottles
            totalBottles += newFullBottles
        return totalBottles

File: test_data/code/java/1518-water-bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int totalBottles = numBottles;
        int emptyBottles = numBottles;
        
        while (emptyBottles >= numExchange) {
            int newFullBottles = emptyBottles / numExchange;
            totalBottles += newFullBottles;
            emptyBottles = newFullBottles + (emptyBottles % numExchange);
        }
        
        return totalBottles;
    }
}